### DISCUSSION TITLE
Fire general functions in BaseController

### DISCUSSION THREAD

**Discussion by kriscr** [pimcore/pimcore#10866] [Q&A / Support]
I was wondering how to get general functions in your BaseController that can be used in other controllers?
In Pimcore 6 

```
    public function onKernelController(FilterControllerEvent $event)
    {
        /** @var $this->truncate \AppBundle\Templating\Helper\CheckElement */
        $this->checkelement = $this->get('AppBundle\Templating\Helper\CheckElement');
    }
```

What do I need to do to migrate to Pimcore 10?

**Comment by M-Ali-Aziz** [comment]
I have the same problem.
Unfortunately I can’t find anything about that on Pimcore documentation page.

**Reply by kriscr** [reply]
I found a solution to move all my template helpers to Twig Extension.
More info on https://symfony.com/doc/current/templating/twig_extension.html

**Reply by M-Ali-Aziz** [reply]
Thank you @kriscr 
This is very good solution when it comes to template helpers,
but `onKernelController()` function in the BaseController on Pimcore 6 was a kind of a `__construct()` function for the BaseController class were you could have properties available on all controllers class that extends the  BaseController class.

And this is something I'm missing and don’t know how to migrate to Pimcore 10.

**Reply by kriscr** [reply]
Yes, I know.

I use __construct on almost every controller and create a specific services so I could get the paramaters I want to use.
You can reuse the service depending on the use in your controller.

Example:
`    protected $translator;

    public function __construct(Translator $translator)
    {
        $this->translator = $translator;
    }`

**Reply by M-Ali-Aziz** [reply]
Yes, there is no problem to use `__construct()` function on a Service on Pimcore 6/Symfony 4 and then use the service in every controller I need.
But my question is if there is any way to make  BaseController class with a `__construct()` function and override the current `onKernelController()` function when i migrate from Pimcore 6 to Pimcore 10?

**Reply by kriscr** [reply]
In Pimcore 10, you need to use `__construct()` to call the services you needed in every controller. This was the way I migrate everything from Pimcore 6 to 10. I'm only using the BaseController general functions.

**Reply by M-Ali-Aziz** [reply]
Ok, so I need to update to Pimcore 10 first and then override the BaseController `onKernelController()` function to `__construct()` function !?

**Reply by kriscr** [reply]
No, you need to add `__construct` to every controller and add the services you need so you can pass certain parameters.
The BaseController holds only general functions and doesn't pass any parameters.
It will be more work to rewrite your code. This way, it's easier to reuse the services and only load the services you need per controller.

**Reply by M-Ali-Aziz** [reply]
Ok, Thank you for your reply :)
So it is possible to add `__construct ` function to every Class Controller on Pimcore 10 (ex. ContentController), this is not possible on Pimcore 6.
But any way I need to update to Pimcore 10 first then I'll test your solution.

Thank you @kriscr :)
